app/lib/actions/katello/host/hypervisors_update.rb

```diff
@@ -0,0 +1,62 @@
+module Actions
+  module Katello
+    module Host
+      class HypervisorsUpdate < Actions::EntryAction
+        middleware.use ::Actions::Middleware::RemoteAction
+
+        def plan(environment, content_view, hypervisor_results)
+          plan_self(:environment_id => environment.id, :content_view_id => content_view.id,
+                    :hypervisor_results => hypervisor_results)
+        end
+
+        def finalize
+          environment = ::Katello::KTEnvironment.find(input[:environment_id])
+          content_view = ::Katello::ContentView.find(input[:content_view_id])
+          hypervisor_results = input[:hypervisor_results]
+
+          %w(created updated unchanged).each do |group|
+            if hypervisor_results[group]
+              hypervisor_results[group].each do |hypervisor|
+                update_or_create_hypervisor(environment, content_view, hypervisor)
+              end
+            end
+          end
+        end
+
+        def update_or_create_hypervisor(environment, content_view, hypervisor_json)
+          name = hypervisor_json[:name]
+
+          # Since host names must be unique yet hypervisors may have unique subscription
+          # aspects in different orgs
+          host = ::Host.find_by_name(name)
+          if host
+            if host.organization.id != content_view.organization.id
+              name = "virt-who-#{name}-#{content_view.organization.id}"
+            end
+          end
+
+          host = ::Katello::Host::SubscriptionAspect.find_or_create_host_for_hypervisor(name,
+                                    content_view.organization)
+          host.subscription_aspect ||= ::Katello::Host::SubscriptionAspect.new
+          host.subscription_aspect.update_from_consumer_attributes(hypervisor_json)
+          host.subscription_aspect.update_attributes(:uuid => hypervisor_json[:uuid])
+          host.subscription_aspect.save!
+
+          # TODO: Remove this legacy
```

Comment:
I'm good with keeping the todo just add the issue in the comment.

Eric
On Nov 20, 2015 2:58 PM, "Justin Sherrill" <notifications@github.com> wrote:

> In app/lib/actions/katello/host/hypervisors_update.rb
> <https://github.com/Katello/katello/pull/5573#discussion_r45511329>:
>
> > +          # aspects in different orgs
> > +          host = ::Host.find_by_name(name)
> > +          if host
> > +            if host.organization.id != content_view.organization.id
> > +              name = "virt-who-#{name}-#{content_view.organization.id}"
> > +            end
> > +          end
> > +
> > +          host = ::Katello::Host::SubscriptionAspect.find_or_create_host_for_hypervisor(name,
> > +                                    content_view.organization)
> > +          host.subscription_aspect ||= ::Katello::Host::SubscriptionAspect.new
> > +          host.subscription_aspect.update_from_consumer_attributes(hypervisor_json)
> > +          host.subscription_aspect.update_attributes(:uuid => hypervisor_json[:uuid])
> > +          host.subscription_aspect.save!
> > +
> > +          # TODO: Remove this legacy
>
> or possibly keep it to make it easier to 'find' what to delete?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5573/files#r45511329>.
>